src/modules/IscsiClientLib.rb

```diff
@@ -819,6 +817,31 @@ def deleteRecord
       ret
     end
 
+    # Delete the current node from the database of discovered targets
+    #
+    # It does not check whether the node is connected. Bear in mind iscsiadm manpage states the
+    # following. "Delete should not be used on a running session. If it is iscsiadm will stop the
+    # session and then delete the record."
```

Comment:
Actually, I jus tested this. If you try to delete a node upon which a session is based, iscsiadm will refuse the request, leaving the Node entry in the Database intact.